File: communicationAPI/src/main/java/eu/cybergeiger/communication/PluginListener.java
package eu.cybergeiger.communication;

/**
 * <p>Interface for plugins listening for events.</p>
 */
public interface PluginListener {

  /**
   * <p>Listener for Geiger events.</p>
   *
   * <p>Any plugin must register for such events and </p>
   *
   * @param url the url the message was sent to
   * @param msg the message
   */
  void pluginEvent(GeigerUrl url, Message msg);

  public byte[] toByteArray();
}

File: lib/src/plugin/plugin_listener.dart
library geiger_api;

import '../message/geiger_url.dart';
import '../message/message.dart';

/// Interface for plugins listening for events.
abstract class PluginListener {
  /// Called when the listened event sends a [msg] to a [url].
  void pluginEvent(GeigerUrl? url, Message msg);
}
